include/ur_api.h

```diff
@@ -1961,6 +1963,36 @@ urTearDown(
     void* pParams                                   ///< [in] pointer to tear down parameters
     );
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Retrieve string representation of the underlying adapter specific
+///        result reported by the the last API that returned
+///        UR_RESULT_ADAPTER_SPECIFIC. Allows for an adapter independent way to
+///        return an adapter specific result.
+/// 
+/// @details
+///     - The string returned via the pMessage is a NULL terminated C style
+///       string.
+///     - The string returned via the pMessage is thread local.
+///     - The entry point will return UR_RESULT_SUCCESS if the result being
+///       reported is to be considered a warning. Any other result code returned
+///       indicates that the adapter specific result is an error.
+///     - The memory in the string returned via the pMessage is owned by the
+///       adapter.
+///     - The application may call this function from simultaneous threads.
+///     - The implementation of this function should be lock-free.
```

Comment:
I think this should talk about using thread local storage explicitly in addition to these general statements about thread safety.